### DISCUSSION TITLE
type WithoutCallSignature<T>

### DISCUSSION THREAD

**Discussion by xeinebiu** [pmndrs/zustand#2370] [Q&A]
### Issue

https://github.com/pmndrs/zustand/blob/827d5aacafa1974fff012955ea6f24fde51dc7ef/src/context.ts#L35

What does exactly this type does. For my understanding, it returns the exact same type as provided.

### Suggestion

To match the name of the type, it implementation should be as
```typescript
type WithoutCallSignature<T> = {
  [K in keyof T as T[K] extends Function ? never : K]: T[K];
};
```

**Comment by dai-shi** [comment]
This is what we want: https://tsplay.dev/wOK1lw

<img width="462" alt="image" src="https://github.com/pmndrs/zustand/assets/490574/03a8a8b7-b760-43be-a61f-0c64b62483a2">

**Reply by xeinebiu** [reply]
Thanks for the clearification. 

I confused it with the idea that it removes all functions from a type, leaving only with types like object, array or primitive types.
